### ISSUE TITLE
mpirun `--do-not-launch` flag causes request to file bug report

### ISSUE THREAD

**Issue by christoph-conrads** [open-mpi/ompi#8292]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

(ISA is x86-64 unless shown otherwise)

| OS | OpenMPI Version | Exhibits problem? |
| -- | -- | -- |
| CentOS 8 | 4.0.2 | yes |
| Debian 9 (i386) | 2.0.2 |
| Debian 9 | 2.0.2 |
| Debian 10 | 3.1.3 | yes |
| Ubuntu 18 | 2.1.1 |
| Ubuntu 20 | 4.0.3 | yes |

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Installation with distribution's package manager (apt on Ubuntu and Debian, dnf on CentOS)

### Please describe the system on which you are running

see the table above

## Details of the problem

The `mpirun` flag `--do-not-launch` causes unrequested output and shows a request to file a bug report.

The expected behavior is a successful mpirun exit if all parameters on the command line are correct without execution of the program given at the end of the command line. Example from Ubuntu 18:
```
$ lsb_release --all
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic
$ mpirun --version
mpirun (Open MPI) 2.1.1

Report bugs to http://www.open-mpi.org/community/help/
$ mpirun --do-not-launch -n 1 echo foo
$ 
```
With OpenMPI version 3, mpirun stills exits successfully but it prints unrequested text. Output on Debian 10:
```
$ lsb_release --all
bash: lsb_release: command not found
$ uname -a
Linux debian-10 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 GNU/Linux
$ mpirun --version
mpirun (Open MPI) 3.1.3

Report bugs to http://www.open-mpi.org/community/help/
$ mpirun --do-not-launch -n 1 echo foo && echo success
[debian-10:00986] LAUNCH MSG RAW SIZE: 472
--------------------------------------------------------------------------
An internal error has occurred in ORTE:

[[51802,0],0] FORCE-TERMINATE AT (null):0 - error base/plm_base_launch_support.c(573)

This is something that should be reported to the developers.
--------------------------------------------------------------------------
success
$
```
With OpenMPI version 4, mpirun stills exits successfully but it prints more unrequested text. Output on Ubuntu 20:
```
$ lsb_release --all
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
$ mpirun --version
mpirun (Open MPI) 4.0.3

Report bugs to http://www.open-mpi.org/community/help/
$ mpirun --do-not-launch -n 1 echo foo && echo success
 Data for JOB [65390,1] offset 0 Total slots allocated 1

 ========================   JOB MAP   ========================

 Data for node: ubuntu-20	Num slots: 1	Max slots: 0	Num procs: 1
 	Process OMPI jobid: [65390,1] App: 0 Process rank: 0 Bound: socket 0[core 0[hwt 0-1]]:[BB/../../../../..]

 =============================================================
[ubuntu-20:02910] LAUNCH MSG RAW SIZE: 465
--------------------------------------------------------------------------
An internal error has occurred in ORTE:

[[65390,0],0] FORCE-TERMINATE AT (null):0 - error base/plm_base_launch_support.c(593)

This is something that should be reported to the developers.
--------------------------------------------------------------------------
success
```

**Comment by rhc54** [comment]
This is fixed on the master branch (soon to be v5.0) - not sure if someone will backport such things to the 4.x series.

**Comment by rhc54** [comment]
OMPI v5.0 should be coming out in the near future. In the interim, you are welcome to use a nightly tarball from either the master of v5.0 branch (https://www.open-mpi.org/nightly/)
